osde2e/route_monitor_operator_tests.go

```diff
@@ -0,0 +1,244 @@
+		serviceMonitorsClient = dynamicClient.Resource(schema.GroupVersionResource{
+			Group:    "monitoring.openshift.io",
+			Version:  "v1alpha1",
+			Resource: "routemonitors",
+		}).Namespace(namespace)
+		prometheusRulesClient = dynamicClient.Resource(schema.GroupVersionResource{
+			Group:    "monitoring.coreos.com ",
+			Version:  "v1",
+			Resource: "prometheusrule",
+		}).Namespace(namespace)
+	})
+
+	It("is installed", func(ctx context.Context) {
+		By("checking the namespace exists")
+		err := k8s.Get(ctx, namespace, "", &corev1.Namespace{})
+		Expect(err).ShouldNot(HaveOccurred(), "namespace %s not found", namespace)
+
+		By("checking the deployment exists and is available")
+		EventuallyDeployment(ctx, k8s, deploymentName, namespace).Should(BeAvailable())
+	})
+
+	It("can be upgraded", func(ctx context.Context) {
+		By("forcing operator upgrade")
+		err := k8s.UpgradeOperator(ctx, operatorName, namespace)
+		Expect(err).ShouldNot(HaveOccurred(), "operator upgrade failed")
+	})
+
+	It("has all of the required resources", func(ctx context.Context) {
+		_, err := serviceMonitorsClient.Get(ctx, consoleName, metav1.GetOptions{})
+		Expect(err).ShouldNot(HaveOccurred(), "Could not get console serviceMonitor")
+
+		_, err = prometheusRulesClient.Get(ctx, consoleName, metav1.GetOptions{})
+		Expect(err).ShouldNot(HaveOccurred(), "Could not get console prometheusRule")
+	})
+
+	It("required dependent resources are created", func(ctx context.Context) {
+		By("Creating a pod, service and route to monitor with a ServiceMonitor and PrometheusRule")
+		var (
+			pod      *corev1.Pod
+			svc      *corev1.Service
+			appRoute *routev1.Route
+			rmo      *routemonitorv1alpha1.RouteMonitor
+		)
+		By("Creating the test pod")
+		pod = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      routeMonitorName,
+				Namespace: namespace,
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  "test",
+						Image: "quay.io/jitesoft/nginx:mainline",
+					},
+				},
+			},
+		}
+		err := k8s.Create(ctx, pod)
+		Expect(err).ShouldNot(HaveOccurred(), "Could not create a pod")
+
+		By("Checking the pod state")
+		var phase kubev1.PodPhase
+		err = wait.PollUntilContextTimeout(ctx, 15*time.Second, 2*time.Minute, false, func(ctx context.Context) (bool, error) {
+			err := k8s.Get(ctx, pod.Name, pod.Namespace, pod)
+			Expect(err).ShouldNot(HaveOccurred(), "Couldn't get the testing pod")
+			if pod != nil {
+				phase = pod.Status.Phase
+				if phase == kubev1.PodRunning || phase == kubev1.PodFailed {
+					return true, nil
+				}
+			}
+			GinkgoLogr.Info(fmt.Sprintf("Waiting for Pod '%s/%s' to be %s, currently %s...", pod.Namespace, pod.Name, kubev1.PodRunning, phase))
+			return false, nil
+		})
+		Expect(phase).To(Equal(kubev1.PodRunning), fmt.Sprintf("pod %s in ns %s is not running, current pod state is %v", routeMonitorName, namespace, phase))
```

Comment:
```suggestion
			return phase == kubev1.PodRunning, nil
		})
		Expect(err).ShouldNot(HaveOccurred(), fmt.Sprintf("pod %s in ns %s is not running, current pod state is %v", routeMonitorName, namespace, phase))
```

this way we will poll _until_ the pod is running, and assert that there was no error while waiting